common/persistence/sql/sqlplugin/postgres/plugin.go

```diff
@@ -86,29 +83,46 @@ func (d *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
 	}
 
 	dbName := cfg.DatabaseName
+	db, err := sqlx.Connect(PluginName, buildDSN(cfg, dbName, host, port))
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Maps struct names in CamelCase to snake without need for db struct tags.
+	db.MapperFunc(strcase.ToSnake)
+	return db, nil
+}
+
+func buildDSN(cfg *config.SQL, dbName string, host string, port string) string {
 	//NOTE: postgres doesn't allow to connect with empty dbName, the admin dbName is "postgres"
 	if dbName == "" {
 		dbName = "postgres"
 	}
-	password := ""
+	params := url.Values{}
+	attrs := ""
+	
 	if cfg.Password != "" {
-		password = fmt.Sprintf(dataSourceNamePostgresPassword, cfg.Password)
+		params.Add("password", cfg.Password)
 	}
-	db, err := sqlx.Connect(PluginName, fmt.Sprintf(dataSourceNamePostgres, cfg.User, host, port, dbName, password))
 
-	if err != nil {
-		return nil, err
+	if cfg.TLS != nil && cfg.TLS.Enabled {
+		params.Add("ssl", "true")
+		params.Add("sslmode", "require")
+		params.Add("sslrootcert", cfg.TLS.CaFile)
+		params.Add("sslkey", cfg.TLS.KeyFile)
+		params.Add("sslcert", cfg.TLS.CertFile)
 	}
+	attrs = params.Encode()
 
-	// Maps struct names in CamelCase to snake without need for db struct tags.
-	db.MapperFunc(strcase.ToSnake)
-	return db, nil
+	dsn := fmt.Sprintf(dsnFmt, "postgresql", cfg.User, host, port, dbName)
```

Comment:
Tests are failling. Seems like this is not compatible with pq server that without TSL enabled:
```

FAIL: TestSQLHistoryV2PersistenceSuite (0.00s) | 1s
-- | --
  | panic: pq: SSL is not enabled on the server [recovered]
  | panic: pq: SSL is not enabled on the server
  |  
  | goroutine 10 [running]:
  | testing.tRunner.func1(0xc000392300)
  | /usr/local/go/src/testing/testing.go:874 +0x69f
  | panic(0x2030fe0, 0xc0000b8880)
  | /usr/local/go/src/runtime/panic.go:679 +0x1b2
  | github.com/uber/cadence/common/persistence/sql.(*TestCluster).CreateDatabase(0xc0005a4840)
  | /cadence/common/persistence/sql/sqlPersistenceTest.go:113 +0x2df
  | github.com/uber/cadence/common/persistence/sql.(*TestCluster).SetupTestDatabase(0xc0005a4840)
  | /cadence/common/persistence/sql/sqlPersistenceTest.go:74 +0x51
  | github.com/uber/cadence/common/persistence/persistence-tests.(*TestBase).Setup(0xc0003c2658)
  | /cadence/common/persistence/persistence-tests/persistenceTestBase.go:178 +0xdf
  | github.com/uber/cadence/common/persistence/sql/sqlplugin/postgres.TestSQLHistoryV2PersistenceSuite(0xc000392300)
  | /cadence/common/persistence/sql/sqlplugin/postgres/postgres_server_test.go:68 +0xed
  | testing.tRunner(0xc000392300, 0x22e8f98)
  | /usr/local/go/src/testing/testing.go:909 +0x19a
  | created by testing.(*T).Run
  | /usr/local/go/src/testing/testing.go:960 +0x652
  | FAIL	github.com/uber/cadence/common/persistence/sql/sqlplugin/postgres	0.041s
  | FAIL
  | make: *** [Makefile:196: cover_profile] Error 1
  | ⚠️ Failed to run command, exited with 2


```

Reply:
I just made the necessary adjustments, can you please test again?

Reply:
Still not passing:
```

panic: pq: password authentication failed for user "postgres" [recovered]
--
  | panic: pq: password authentication failed for user "postgres"


```
You can see it if you click the `details` in 
`buildkite/cadence-server/pr/golang-unit-test — Failed (exit status 2)`

Reply:
Are you able to run the test locally? That will be easier to debug and fix the test.

Reply:
> yeah e2e tests

Sure since we have it in this repo. The reason we didn't do that is that we ran into timing issues with buildkite. But it should be fine since we are running tests in parallel now.